marlowe-test/src/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -387,18 +386,15 @@ randomRoleNames =
   , "Nonso Ernie Blanka"
   , "Umukoro Alexander Columb"
   , "Urbanus Roland Alison Ty Ryoichi"
-  , "Alcippe Alende Blanka Roland Dafne"  -- NB: Too long for ledger.
   ]
 
 instance Arbitrary Party where
-  arbitrary =
-    do
-       isPubKeyHash <- frequency [(2, pure True), (8, pure False)]
-       if isPubKeyHash
-         then Address <$> arbitrary <*> arbitrary
-         else Role <$> arbitraryFibonacci randomRoleNames
-  shrink (Address _ _) = Role <$> randomRoleNames
-  shrink (Role x)      = Role <$> shrinkByteString (\(TokenName y) -> y) randomRoleNames x
+  arbitrary = frequency
+    [ (1, Address <$> arbitrary <*> arbitrary)
+    , (4, sized \size -> Role <$> elements (takeWhile ((<= size) . fromInteger . lengthOfByteString . unTokenName) randomRoleNames))
+    ]
+  shrink (Address _ _) = []
+  shrink (Role x)      = Role <$> shrinkByteString unTokenName randomRoleNames x
```

Comment:
Fixed improper `shrink` implementation: all values of a `shrink` should be derived from the value being shrunk. Shrinking an address to a random role does not satisfy this.